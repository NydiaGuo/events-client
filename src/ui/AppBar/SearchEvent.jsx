import React, { Component, Fragment } from 'react'
import { compose } from 'recompose'
import SearchIcon from '@material-ui/icons/Search'
import { TextField, IconButton } from '@material-ui/core'
import { withStyles } from '@material-ui/core/styles'

import styles from './styles'

class SearchEvent extends Component {

  searchEvents = () => {
    console.log('Searching Events....')
  }
  render() {
    const { classes } = this.props

    return (
      <Fragment>
        <TextField />
        <IconButton aria-label='Add to favorites' onClick={this.searchEvents}>
          <SearchIcon className={classes.folderIcon} />
        </IconButton>
      </Fragment>
    )
  }
}

export default compose(
  withStyles(styles),
)(SearchEvent)
