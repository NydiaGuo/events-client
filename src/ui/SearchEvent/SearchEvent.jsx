import React, { Component, Fragment } from 'react'
import { connect } from 'react-redux'
import { compose } from 'recompose'
import * as searchActions from 'store/actions/search-actions'
import * as searchSelectors from 'store/selectors/search-selectors'
import { Link, Redirect } from 'react-router-dom'
import { TextField } from '@material-ui/core'
import { withStyles } from '@material-ui/core/styles'
import { Button } from '@material-ui/core'
import { green } from '../../logger/index'

export class SearchEvent extends Component {

  constructor(props) {
  super(props)
    this.state = {
      searchText: '',
      showSearchIcon: true,
      search: false,
    }
  }


  handleChange = (value) => {
    green('handleChange', value)
    // this.setState((prevState, props) => ({
    //   searchText: value
    // }))
    this.props.setSearchText(value)
  }

  searchEvents = () => {
    green('searchEvent', 'hello')
    this.setState({ search: true })
  }

  render() {
    const { classes } = this.props
    const { searchText } = this.props

    green('Search: render()')
    green('searchText', searchText.length)
    green('props', this.props)

    // if (search) {
    //   return <Redirect to={`/events/search/${searchText}`} />
    // }
    return (
      <Fragment>
        <TextField onChange={e => this.handleChange(e.target.value)} value={searchText} />

        <Button
            className={classes.searchIcon}
            onClick={this.searchEvents}
        >
          Search
        </Button>

      </Fragment>
    )
  }
}

const styles = {

}

const mapStateToProps = (state) => {
  const o = searchSelectors.getSearchText(state)
  green('o', o)
  return {
    searchText: o
  }

}


export default compose(
  withStyles(styles),
  connect(mapStateToProps, searchActions)
)(SearchEvent)


