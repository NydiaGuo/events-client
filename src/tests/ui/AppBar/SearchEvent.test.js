import React from 'react'
import {shallow} from 'enzyme'

import SearchEvent from '../../../ui/AppBar/SearchEvent'

test('should render search correctly', () => {
  const wrapper = shallow(<SearchEvent />)
  expect(wrapper).toMatchSnapshot()
})

