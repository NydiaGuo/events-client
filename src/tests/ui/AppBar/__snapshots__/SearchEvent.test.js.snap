// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render search correctly 1`] = `
<Connect(SearchEvent)
  classes={
    Object {
      "cancelIcon": "Connect-SearchEvent--cancelIcon-2",
      "searchIcon": "Connect-SearchEvent--searchIcon-1",
    }
  }
/>
`;
